### ISSUE TITLE
acquireTokenByClientCredential caching/refreshing behaviour

### ISSUE THREAD

**Issue by scottdurow** [AzureAD/microsoft-authentication-library-for-js#4991]
### Core Library

MSAL Node (@azure/msal-node)

### Core Library Version

1.11.0

### Wrapper Library

Not Applicable

### Wrapper Library Version

None

### Description

Is there any way to use acquireTokenByClientCredential (supplying a client id/secret), cache a refresh token and then allow calling acquireTokenSilent to refresh after that?

This would support a scenario where I want to avoid storing the client secret (supplying it once during setting up a tool) - and rely on the cached refresh token from that point. This would be logically similar to the scenario where an interactive login is performed first, and then the token is refreshed subsequently using acquireTokenSilent - except with the client id and secret are used instead of the interactive login.

Thank you!

### MSAL Configuration

```javascript
{
    auth: {
      clientId: clientId,
      authority: `https://login.windows.net/${tenantID}`,
      clientSecret: clientSecret,
    },
    cache: {
      cachePlugin: MsalCachePlugin,
    },
    system: {
      loggerOptions: {
        loggerCallback: logger,
        logLevel: logger ? LogLevel.Verbose : undefined,
        piiLoggingEnabled: true,
      },
    },
  }
```

### Identity Provider

Azure AD / MSA

### Source

External (Customer)

**Comment by sameerag** [comment]
@scottdurow Apologies for the delay in the response.
`refresh_token` can never be returned for `client_credentials_flow` by design as per the spec [here](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.3).

The USP of`client_credentials` is to be able to revoke the credentials as needed without worrying about stolen refresh tokens. This is highlighted in the spec [here](https://datatracker.ietf.org/doc/html/rfc6749#section-3.2.1) as follows:

      Recovering from a compromised client by disabling the client or
      changing its credentials, thus preventing an attacker from abusing
      stolen refresh tokens.  Changing a single set of client
      credentials is significantly faster than revoking an entire set of
      refresh tokens.

**Comment by scottdurow** [comment]
Ok thanks @sameerag - so in my scenario I would need to store the client secret somewhere since the msal-node extensions don't allow you to store them in the same way as the refreshtoken.

**Comment by sameerag** [comment]
Yes. cc @jmprieur in case he has anything to add.

**Comment by jmprieur** [comment]
@scottdurow if you cannot use a secret store, you could use Azure KeyVault.
See [here](https://www.bing.com/search?q=node.js%20secret%20store&qs=n&form=QBRE&=Search%20%7B0%7D%20for%20%7B1%7D&=Search%20work%20for%20%7B0%7D&=%25eManage%20Your%20Search%20History%25E&sp=-1&pq=node.js%20secret%20store&sc=0-20&sk=&cvid=73D63A1F531F48C6870214B31943D6AC&ghsh=0&ghacc=0&ghpl=)

**Comment by sameerag** [comment]
Thanks @jmprieur. Closing the issue @scottdurow, please reach out in case you are blocked on anything else with a new issue.
